versions.tf

```diff
@@ -1,3 +1,20 @@
 terraform {
-  required_version = ">= 0.12.14"
+  required_version = "~> 0.12.28"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
```

Comment:
whyyyyyyyyyy

Reply:
because it's better in every way, with just a few breaking changes that don't matter in this context.

Reply:
```suggestion
      version = ">= 2.45"
```